lib/ansible/modules/cloud/amazon/ec2_vpc_vgw_facts.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/python
+#
+# This is a free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This Ansible library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+# import module snippets
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import ec2_argument_spec, get_aws_connection_info
+from ansible.module_utils.ec2 import boto3_conn, camel_dict_to_snake_dict
+from ansible.module_utils.ec2 import ansible_dict_to_boto3_filter_list, HAS_BOTO3
+
+
+DOCUMENTATION = '''
+---
+module: ec2_vpc_vgw_facts
+short_description: Gather facts about virtual gateways in AWS
+description:
+    - Gather facts about virtual gateways in AWS.
+version_added: "2.3"
+requirements: [ boto3 ]
+options:
+  filters:
+    description:
+      - A dict of filters to apply. Each dict item consists of a filter key and a filter value. See U(http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeRouteTables.html) for possible filters.
+    required: false
+    default: None
+  vpn_gateway_ids:
+    description:
+      - Get details of a specific Virtual Gateway ID. This value should be provided as a list. 
+    required: false
+    default: None
+author: "Nick Aslanidis (@naslanidis)"
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# # Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: Gather facts about all virtual gateways for an account or profile
+  ec2_vpc_vgw_facts:
+    region: ap-southeast-2
+    profile: production
+  register: vgw_facts
+
+- name: Gather facts about a filtered list of Virtual Gateways
+  ec2_vpc_vgw_facts:
+    region: ap-southeast-2
+    profile: production
+    filters:
+        "tag:Name": "vgw-123"
```

Comment:
Can you use a different example here? It's easy to confuse this with a dummy VGW ID. Maybe "main-virt-gateway" would be a better docs slug.